OSDOCS-10891:dedicated troubleshooting section for data cleanup (#80106)
openshift/openshift-docs: modules/microshift-data-cleaning-container-images.adoc

// Module included in the following assemblies:
//
// * microshift_troubleshooting/microshift-cleanup-data.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-data-cleaning-container-images_{context}"]
= Cleaning all data keeping the container images

Run the following command to skip deleting the container image when cleaning all data. This helps speed up {microshift-short} restart after data cleanup as the necessary container images are already present locally when starting the service.This command performs the following cleanup actions:

* Stop and disable {microshift-short} services
* Delete all {microshift-short} pods
* Reset network configuration
* Delete the /var/lib/microshift data directory
* Delete OVN configuration

.Procedure

> . Run the following command to skip deleting the container image when cleaning all data.

Reviewer: It's repeated.

Author: done. modified the line

1) Yes
2) No
#? Yes
Stopping MicroShift services
Disabling MicroShift services
Removing MicroShift pods
Deleting the br-int interface
Killing conmon, pause and OVN processes
Removing MicroShift configuration
Removing OVN configuration
MicroShift service was stopped
MicroShift service was disabled
Cleanup succeeded

$ sudo crictl images | awk '{print $1}'
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
registry.redhat.io/lvms4/topolvm-rhel9
registry.redhat.io/openshift4/ose-csi-external-provisioner
registry.redhat.io/openshift4/ose-csi-external-resizer
registry.redhat.io/openshift4/ose-csi-livenessprobe
registry.redhat.io/openshift4/ose-csi-node-driver-registrar
registry.redhat.io/ubi9
----

> . Run the following command to restart the {microshift-short} service.

Reviewer: Please, add the full text from the doc

Author: done. note is added

// Module included in the following assemblies:
//
// * microshift_troubleshooting/microshift-cleanup-data.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-data-cleaning-container-images_{context}"]
= Cleaning all data and keeping the container images

> Retain the container image while cleaning all data by running the following script. This helps speed up {microshift-short} restart after data cleanup as the necessary container images are already present locally when starting the service. This script performs the following cleanup actions:

Reviewer: ```suggestion
Retain the MicroShift container images while cleaning all data by running the following script. This helps speed up {microshift-short} restart after data cleanup as the necessary container images are already present locally when starting the service. This script performs the following cleanup actions:
```

Author: done

> Retain the {microshift-short} container images while cleaning all data by running the following script. This helps speed up {microshift-short} restart after data cleanup as the necessary container images are already present locally when starting the service. This script performs the following cleanup actions:

Reviewer: Yes, we need to stress what options need to be used for the current action.

* Stop and disable {microshift-short} services
* Delete all {microshift-short} pods
* Reset network configuration
* Delete the /var/lib/microshift data directory

> * Delete OVN configuration

Reviewer: Yes, but's it's specific to OVN.

.Procedure

> . Run the following script.

Reviewer: ```suggestion
. Run the following command.
```

Author: done

> . Run the following command.

Reviewer 2: same comment here; we need to adhere to docs style

Author: done

> . Execute the `microshift-cleanup-data` script, by running the following command.

Reviewer 2: Update per command structure noted previously.

+
[WARNING]
====

> This script deletes all {microshift-short} and user-specific data. Use with caution.

Reviewer 2: ```suggestion
This option deletes all {microshift-short} and user workloads. Use with caution.
```

Killing conmon, pause and OVN processes
Removing MicroShift configuration
Removing OVN configuration
MicroShift service was stopped
MicroShift service was disabled
Cleanup succeeded

$ sudo crictl images | awk '{print $1}'
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
registry.redhat.io/lvms4/topolvm-rhel9
registry.redhat.io/openshift4/ose-csi-external-provisioner
registry.redhat.io/openshift4/ose-csi-external-resizer
registry.redhat.io/openshift4/ose-csi-livenessprobe
registry.redhat.io/openshift4/ose-csi-node-driver-registrar
registry.redhat.io/ubi9
----
+
[NOTE]
====
The {microshift-short} service is stopped and disabled after executing the script.
====
. Restart the {microshift-short} service by running the following command.
+
[source,terminal]
----

> $ sudo systemctl enable --now microshift

Reviewer: There are two types of commands: one when microshift is only stopped, and another when it's also disabled.
The document has all this correctly.

// Module included in the following assemblies:
//
// * microshift_troubleshooting/microshift-cleanup-data.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-data-cleaning-container-images_{context}"]

> = Cleaning all {microshift-short} data and keeping the container images

Reviewer: ```suggestion
= Cleaning all data and keeping the container images
```

> Retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` arguments.

Reviewer 3: Should this be phrased, similar to other sections on this page and throughout the docs, as something the user can do?
```suggestion
You can retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` arguments.
```

> Retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `-all` and `--keep-images` arguments.

Reviewer: ```suggestion
Retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` arguments.
```

Keeping the container images helps speed up {microshift-short} restart after data cleanup because the necessary container images are already present locally when you start the service.

> When you run the script with the `-all` and `--keep-images` arguments, you perform the following cleanup actions:

Reviewer: ```suggestion
When you run the script with the `--all` and `--keep-images` arguments, you perform the following cleanup actions:
```

> * All {microshift-short} services are stopped and disabled
> * Deletes all {microshift-short} pods
> * Resets network configuration
> * Deletes the `/var/lib/microshift` data directory
> * Deletes OVN-K networking configuration

Reviewer 3: Same comment as the other module about this being a list of actions that "you" the user do, up to you if and how you'd want to change this:
```suggestion
* Stop and disable all {microshift-short} services
* Delete all {microshift-short} pods
* Reset network configuration
* Delete the `/var/lib/microshift` data directory
* Delete OVN-K networking configuration
```

[WARNING]
====
This option deletes all {microshift-short} data and user workloads. Use with caution.
====

.Prerequisites

> . You are logged into the {microshift-short} as an administrator with root-user access.
> . You have filed a support case.

Reviewer 3: Same comment about using an unordered list:
```suggestion
* You are logged into the {microshift-short} as an administrator with root-user access.
* You have filed a support case.
```

.Procedure

> . Cleanup all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:

Reviewer 3: Same comment about splitting cleanup into two words:
```suggestion
. Clean up all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:
```

> . Cleanup all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:

Reviewer 3: I think since you say "by running the following command", and since the source block below already states what is part of the command, you can remove this part of the step:
```suggestion
. Cleanup all data and user workloads while retaining the {microshift-short} container images by entering the following command:
```

+
[source,terminal]
----
$ sudo microshift-cleanup-data --all --keep-images
----
+
.Example output
[source,terminal]
----
DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?
1) Yes
2) No
#? Yes
Stopping MicroShift services
Disabling MicroShift services
Removing MicroShift pods
Deleting the br-int interface
Killing conmon, pause and OVN processes
Removing MicroShift configuration
Removing OVN configuration
MicroShift service was stopped
MicroShift service was disabled
Cleanup succeeded

> $ sudo crictl images | awk '{print $1}'

Reviewer 3: Just checking - is this legitimately part of the script output or is this an a command that users need to run?

Reviewer: We could add something like:
Verify that the container images are still present after running the `microshift-cleanup-data` script.

```
$ sudo crictl images | awk '{print $1}'
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
...
...
```

> 
> $ sudo crictl images | awk '{print $1}'
> IMAGE
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> quay.io/openshift-release-dev/ocp-v4.0-art-dev
> registry.redhat.io/lvms4/topolvm-rhel9
> registry.redhat.io/openshift4/ose-csi-external-provisioner
> registry.redhat.io/openshift4/ose-csi-external-resizer
> registry.redhat.io/openshift4/ose-csi-livenessprobe
> registry.redhat.io/openshift4/ose-csi-node-driver-registrar
> registry.redhat.io/ubi9
> ----
> +
> [IMPORTANT]
> ====
> The {microshift-short} service is stopped and disabled after you run the script.
> ====

Reviewer 3: Based on the SME's comment [here](https://github.com/openshift/openshift-docs/pull/80106#discussion_r1732778861), the second part of this output actually should be split into its own step, like this:
```suggestion
----
+
[IMPORTANT]
====
The {microshift-short} service is stopped and disabled after you run the script.
====

. Verify that the container images are still present by running the following command:
+
[source,terminal]
----
$ sudo crictl images | awk '{print $1}'
----
+
.Example output
[source,terminal]
----
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
registry.redhat.io/lvms4/topolvm-rhel9
registry.redhat.io/openshift4/ose-csi-external-provisioner
registry.redhat.io/openshift4/ose-csi-external-resizer
registry.redhat.io/openshift4/ose-csi-livenessprobe
registry.redhat.io/openshift4/ose-csi-node-driver-registrar
registry.redhat.io/ubi9
----

```

. Restart the {microshift-short} service by running the following command:
+
[source,terminal]
----

> $ sudo systemctl enable --now microshift

Reviewer 4: I don'y think this is the correct command to restart microshift. Can you confirm @ggiguash 

sudo systemctl restart microshift

```suggestion
$ sudo systemctl enable --now microshift
```

* Delete OVN-K networking configuration

[WARNING]
====
This option deletes all {microshift-short} data and user workloads. Use with caution.
====

.Prerequisites
* You are logged into {microshift-short} as an administrator with root-user access.
* You have filed a support case.

.Procedure

. Clean up all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:
+
[source,terminal]
----
$ sudo microshift-cleanup-data --all --keep-images
----
+
.Example output
[source,terminal]
----
DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?
1) Yes
2) No
#? Yes
Stopping MicroShift services
Disabling MicroShift services
Removing MicroShift pods
Deleting the br-int interface
Killing conmon, pause and OVN processes
Removing MicroShift configuration
Removing OVN configuration
MicroShift service was stopped
MicroShift service was disabled
Cleanup succeeded
----
+
[IMPORTANT]
====
The {microshift-short} service is stopped and disabled after you run the script.

> ====

Reviewer: This belongs before line 90

----
+
.Example output
[source,terminal]
----
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
registry.redhat.io/lvms4/topolvm-rhel9
registry.redhat.io/openshift4/ose-csi-external-provisioner
registry.redhat.io/openshift4/ose-csi-external-resizer
registry.redhat.io/openshift4/ose-csi-livenessprobe
registry.redhat.io/openshift4/ose-csi-node-driver-registrar
registry.redhat.io/ubi9
----
+
[IMPORTANT]
====
The {microshift-short} service is stopped and disabled after you run the script.
====
. Restart the {microshift-short} service by running the following command:
+
[source,terminal]
----

> $ sudo systemctl enable --now microshift

Reviewer 5: @ggiguash Do we want to point the user to the location of the newly generated kubeconfig files after the procedure?   or is that well-covered elsewhere?

Reviewer: It is documented [here](https://docs.redhat.com/en/documentation/red_hat_build_of_microshift/4.16/html/installing/microshift-install-rpm#accessing-microshift-cluster_microshift-install-rpm)

If it is not "obvious" that deleting the whole database would necessitate update of kubeconfig to access the new cluster, we may want to add it, yes.
